server/src/test/java/org/jetbrains/bsp/bazel/server/sync/languages/LanguagePluginServiceTest.java

```diff
@@ -0,0 +1,417 @@
+package org.jetbrains.bsp.bazel.server.sync.languages;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.vavr.collection.HashSet;
+import io.vavr.collection.Set;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.jetbrains.bsp.bazel.bazelrunner.BasicBazelInfo;
+import org.jetbrains.bsp.bazel.server.sync.BazelPathsResolver;
+import org.jetbrains.bsp.bazel.server.sync.languages.cpp.CppLanguagePlugin;
+import org.jetbrains.bsp.bazel.server.sync.languages.java.JavaLanguagePlugin;
+import org.jetbrains.bsp.bazel.server.sync.languages.java.JdkResolver;
+import org.jetbrains.bsp.bazel.server.sync.languages.java.JdkVersionResolver;
+import org.jetbrains.bsp.bazel.server.sync.languages.scala.ScalaLanguagePlugin;
+import org.jetbrains.bsp.bazel.server.sync.languages.thrift.ThriftLanguagePlugin;
+import org.jetbrains.bsp.bazel.server.sync.model.Language;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+public class LanguagePluginServiceTest {
+
+  @Nested
+  @DisplayName("Tests for the method shouldGetPlugin")
+  class shouldGetPluginTest {
+
+    LanguagePluginsService languagePluginsService;
+
+    @BeforeEach
+    void beforeEach() {
+      var bazelInfo = new BasicBazelInfo("", Paths.get(""));
+      var bazelPathsResolver = new BazelPathsResolver(bazelInfo);
+      var jdkResolver = new JdkResolver(bazelPathsResolver, new JdkVersionResolver());
+      var javaLanguagePlugin = new JavaLanguagePlugin(bazelPathsResolver, jdkResolver, bazelInfo);
+      var scalaLanguagePlugin = new ScalaLanguagePlugin(javaLanguagePlugin, bazelPathsResolver);
+      var cppLanguagePlugin = new CppLanguagePlugin();
+      var thriftLanguagePlugin = new ThriftLanguagePlugin(bazelPathsResolver);
+      languagePluginsService =
+          new LanguagePluginsService(
+              scalaLanguagePlugin, javaLanguagePlugin, cppLanguagePlugin, thriftLanguagePlugin);
+    }
+
+    @Test
+    public void shouldGetPluginForJava() {
+      // given
+      Set<Language> languages = HashSet.of(Language.JAVA);
+
+      // test
```

Comment:
lets keep the convention `//given //when //then`

Reply:
Fixed